ts/IterableInboxEmptyState.tsx

```diff
@@ -1,46 +1,74 @@
 'use strict'
 
 import React from 'react'
-import { View, Text, StyleSheet } from 'react-native'
+import { 
+   View, 
+   Text, 
+   StyleSheet,
+} from 'react-native'
 
 import { IterableInboxCustomizations } from '.'
 
 type emptyStateProps = {
-   customizations: IterableInboxCustomizations
+   customizations: IterableInboxCustomizations,
+   tabBarHeight: number,
+   tabBarPadding: number,
+   navTitleHeight: number,
+   contentWidth: number,
+   height: number,
+   isPortrait: boolean
 }
 
-const IterableInboxEmptyState = ({ customizations } : emptyStateProps) => {
+const IterableInboxEmptyState = ({ 
+   customizations, 
+   tabBarHeight,
+   tabBarPadding, 
+   navTitleHeight,
+   height,
+   isPortrait
+} : emptyStateProps) => {
    const defaultTitle = "No saved messages"
    const defaultBody = "Check again later!"
    
-   const title = customizations.noMessagesTitle
-   const body = customizations.noMessagesBody
+   const emptyStateTitle = customizations.noMessagesTitle
+   const emptyStateBody = customizations.noMessagesBody
+
+   let {
+      container,
+      title,
+      body   
+   } = styles
+
+   container = {...container, height: height - navTitleHeight - tabBarHeight - tabBarPadding}
+
+   if(!isPortrait) {
```

Comment:
This got me thinking, is it really necessary to pass down `isPortrait` to everything? Shouldn't it be dynamically accounted for, rather than on render time?

Reply:
IsPortrait is accounted for dynamically. The useIsPortrait hook calculates if the orientation is portrait on orientation change.